packages/synapse-react-client/src/components/SynapseTable/usePrefetchTableData.ts

```diff
@@ -197,20 +204,45 @@ function usePrefetchUserGroupHeaderData() {
   })
 }
 
+function usePrefetchEntityRestrictionData(entitiesToPrefetch: ReferenceList) {
+  // This information only varies by targetId, so we can dedupe the list in case there are multiple versions of the same file
+  const uniqueEntities = useMemo(
+    () => Array.from(new Set([...entitiesToPrefetch.map(e => e.targetId)])),
+    [entitiesToPrefetch],
+  )
```

Comment:
TODO: move this into the hook because there's no point in requiring all callers to do this